supervisor/resolution/__init__.py

```diff
@@ -21,4 +64,58 @@ def unsupported(self) -> List[UnsupportedReason]:
     @unsupported.setter
     def unsupported(self, reason: UnsupportedReason) -> None:
         """Add a reason for unsupported."""
-        self._unsupported.append(reason)
+        if reason not in self._unsupported:
+            self._unsupported.append(reason)
+
+    def clear(self, collection: str) -> None:
```

Comment:
We are going away from str which is not strict to Enum. So we need to use enum like CollectionType for it. Another question is, why not just call on code `self.issue.clear()`

Reply:
Fair point, clearing should only happen from the resolution manager, so we can clear directly